### PR TITLE
Bug/1093 forbidden chars in many fields

### PR DESCRIPTION
**Pull request by kzangeli**
Implements #1093 

Added check of invalid characters to
Entity, ContextAttribute and Metadata, for API v2

### CODE REVIEW

File: test/functionalTest/cases/0000_xml_parse/xmlParsePostNotifyContext.test [hunk 1]
```diff
@@ -52,9 +52,9 @@ payload='<?xml version="1.0" encoding="UTF-8"?>
                 <value>1110</value>
               </contextMetadata>
               <contextMetadata>
-                <name></name>
-                <type></type>
-                <value></value>
+                <name>id2</name>
```
Comment (fgalan) [inline test/functionalTest/cases/0000_xml_parse/xmlParsePostNotifyContext.test]: XML is now a deprecated feature, so modifications in XML-based .test are not very relevant. However, I wonder why this has changed... there was any bug in the XML metadata rendering?

Comment (kzangeli) [inline test/functionalTest/cases/0000_xml_parse/xmlParsePostNotifyContext.test]: These files that have changed (they had empty fields in metadata) are both input and output data for unit tests. The files used for input had to be changed, as with the new checks (the check method is the same for XML, JSONv1 and JSONv2) the unit test fails if the fields are empty.

Comment (fgalan) [inline test/functionalTest/cases/0000_xml_parse/xmlParsePostNotifyContext.test]: NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**